addons/account/models/account_move.py

```diff
@@ -2124,17 +2124,17 @@ def invert_tags_if_needed(repartition_line, tags):
                 continue
 
             # ==== Map tax repartition lines ====
-            if line_vals.get('tax_repartition_line_id'):
-                # Tax line.
-                invoice_repartition_line = self.env['account.tax.repartition.line'].browse(line_vals['tax_repartition_line_id'])
-                if invoice_repartition_line not in tax_repartition_lines_mapping:
-                    raise UserError(_("It seems that the taxes have been modified since the creation of the journal entry. You should create the credit note manually instead."))
-                refund_repartition_line = tax_repartition_lines_mapping[invoice_repartition_line]
-
-                # Find the right account.
-                if cancel:
-                    account_id = line_vals['account_id']
-                else:
+            # We don't do this mapping for cash basis entries: cancelling a cash basis entry usually happens when unreconciling and invoice,
+            # in which case we always want the reverse entry to totally cancel the original one, keeping the same accounts, tags and repartition lines
+            if not self.tax_cash_basis_rec_id:
```

Comment:
`self` could be multiple records

Reply:
There is an ensure_one at the beginning of this menhir ;p

Reply:
You win... this time!